mvpinternshipapp.client/src/components/sale/SaleDetails.jsx

```diff
@@ -0,0 +1,111 @@
+import { Input, Label, Dropdown } from "semantic-ui-react";
+import { useState, useEffect } from 'react';
+import api from '../../api';
+
+import PropTypes from 'prop-types';
+
+export function SaleDetails({ state, setState, open }) {
+    const [customers, setCustomers] = useState([]);
+    const [products, setProducts] = useState([]);
+    const [stores, setStores] = useState([]);
+
+    useEffect(() => {
+        if (open) {
+            let stillAlive = true;
+            const mapToOptions = (objArray) => {
+                return objArray.map(obj => ({
+                    key: obj.id,
+                    value: obj.id,
+                    text: obj.name,
+                }));
+            };
+            api.getCustomers().then((res) => {
+                if (stillAlive) {
+                    setCustomers(mapToOptions(res.data));
+                }
+            });
+            api.getProducts().then((res) => {
+                if (stillAlive) {
+                    setProducts(mapToOptions(res.data));
+                }
+            });
+            api.getStores().then((res) => {
+                if (stillAlive) {
+                    setStores(mapToOptions(res.data));
+                }
+            });
+            return () => (stillAlive = false);
+        }
+    }, [open])
+
+    const setProduct = (productId) => {
+        const { key, text } = products.find((product) => product.value === productId);
+        if (Number.isInteger(key) && text !== undefined) {
```

Comment:
Looks like you are using this code multiple times:

`if (Number.isInteger(key) && text !== undefined) {`

You can actually create a separate function for this and just call it when you need